server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -788,12 +793,6 @@ protected void runInContext() {
                         }
                     }
                 } else {
-                    //previously vm has ip and nic table has ip address. After vm restart or stop/start
```

Comment:
In my opinion, if we cannot know if the VM still has the IP assigned or not then we should not modify the database. In other words, we should be sure that the VM's NIC does not have an IP assigned before updating the database.